bundle/config/mutator/python/apply_python_mutator.go

```diff
@@ -0,0 +1,148 @@
+package python
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/config"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/dyn"
+	"github.com/databricks/cli/libs/dyn/yamlloader"
+	"github.com/databricks/cli/libs/log"
+	"github.com/databricks/cli/libs/process"
+	"path"
+	"path/filepath"
+)
+
+type phase string
+
+const (
+	ApplyPythonMutatorPhasePreInit phase = "preinit"
+	ApplyPythonMutatorPhaseInit    phase = "init"
+)
+
+type applyPythonMutator struct {
+	Phase phase
+}
+
+func ApplyPythonMutator(phase phase) bundle.Mutator {
+	return &applyPythonMutator{
+		Phase: phase,
+	}
+}
+
+func (m *applyPythonMutator) Name() string {
+	return fmt.Sprintf("ApplyPythonMutator(%s)", m.Phase)
+}
+
+func getExperimental(b *bundle.Bundle) config.Experimental {
+	if b.Config.Experimental == nil {
+		return config.Experimental{}
+	}
+
+	return *b.Config.Experimental
+}
+
+func (m *applyPythonMutator) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	experimental := getExperimental(b)
+
+	if !experimental.EnablePyDABs {
+		log.Debugf(ctx, "'experimental.enable_pydabs' isn't enabled, skipping")
+		return nil
+	}
+
+	if experimental.EnablePyDABs && experimental.VEnv.Path == "" {
+		return diag.Errorf("'experimental.enable_pydabs' can only be used when 'experimental.venv.path' is set")
+	}
+
+	mutatorOutput, err := m.runPythonMutator(ctx, experimental.VEnv.Path, &b.Config)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = m.applyMutatorOutput(&b.Config, mutatorOutput)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func (m *applyPythonMutator) runPythonMutator(ctx context.Context, venvPath string, rootConfig *config.Root) (dyn.Value, error) {
+	args := []string{
+		path.Join(venvPath, "bin", "python3"),
+		"-m",
+		"databricks.bundles.build",
+		"--phase",
+		string(m.Phase),
+	}
+
+	rootConfigJson, err := json.Marshal(rootConfig)
+
+	if err != nil {
+		return dyn.InvalidValue, fmt.Errorf("failed to marshal root config: %w", err)
+	}
+
+	pythonLogger := NewStderrLogger(ctx)
+
+	stdout, err := process.Background(
+		ctx,
+		args,
+		process.WithStdinReader(bytes.NewBuffer(rootConfigJson)),
+		process.WithStderrWriter(pythonLogger),
+	)
```

Comment:
Do you have expectations around the working directory when executing this?

Would be good to comment either way.

Reply:
It should be relative to the bundle root, I've explicitly passed directory to `Process`, so we don't need to add a comment about it